corehq/motech/repeaters/tests/test_dbaccessors.py

```diff
@@ -144,6 +147,83 @@ def test_get_overdue_repeat_record_count(self):
         overdue_count = get_overdue_repeat_record_count()
         self.assertEqual(overdue_count, 1)
 
+    def test_get_paged_repeat_records_by_last_checked(self):
+        today = datetime.utcnow()
+        specific_id = 'some_id'
+
+        def _record(last_checked):
+            return RepeatRecord(
+                domain=self.domain,
+                last_checked=last_checked,
+                succeeded=False,
+                repeater_id=self.other_id,
+                next_check=today,
+                payload_id=self.payload_id_2,
+            )
+
+        two_days_ago = RepeatRecord(
+            domain=self.domain,
+            last_checked=today - timedelta(days=2),
+            succeeded=False,
+            failure_reason='some error',
+            repeater_id=specific_id,
+            next_check=today,
+            payload_id=self.payload_id_2,
+        )
+        on_today = RepeatRecord(
+            domain=self.domain,
+            last_checked=today + timedelta(days=2),
+            succeeded=True,
+            repeater_id=self.other_id,
+            next_check=today,
+            payload_id=self.payload_id_2,
+        )
+        two_days_latter = RepeatRecord(
+            domain=self.domain,
+            last_checked=today,
+            succeeded=False,
+            repeater_id=specific_id,
+            next_check=today,
+            payload_id=self.payload_id_2,
+        )
+        records = [two_days_ago, on_today, two_days_latter]
+        RepeatRecord.bulk_save(records)
+
+        def _get_records(startdate, enddate, state=None, repeater_id=None):
+            return get_paged_repeat_records(
+                self.domain, 0, 10,
+                datespan=DateSpan(startdate, enddate),
+                state=state,
+                repeater_id=repeater_id,
+            )
+
+        # Test only by last_checked
+        self.assertEqual(
+            len(_get_records(today - timedelta(days=3), today + timedelta(days=3))),
+            3
+        )
+        self.assertEqual(
+            len(_get_records(today - timedelta(days=1), today + timedelta(days=1))),
+            1
+        )
+        # Test state as well
+        self.assertEqual(
+            len(_get_records(today - timedelta(days=3), today + timedelta(days=3), state='FAIL')),
+            1
+        )
+        # Test repeater_id as well
+        self.assertEqual(
+            len(_get_records(today - timedelta(days=0), today + timedelta(days=2), repeater_id=specific_id)),
+            1
+        )
+        # Test state and repeater_id
+        self.assertEqual(
+            len(_get_records(today - timedelta(days=3),
+                today + timedelta(days=3), state='FAIL', repeater_id=specific_id)),
+            1
+        )
+        RepeatRecord.bulk_delete(records)
```

Comment:
this should be a class cleanup otherwise if a test fails it won't run

Reply:
This is true, but it's only a concern when there is a regression failure on travis and since the test failed it needs to be looked anyway. Otherwise, it's only a concern during the development of this test which I already finished. I kind of went back n forth on adding this. 

I changed it as you suggested, but I am curious to hear if there is really a value.

Reply:
It definitely has value since failures locally can be very hard to debug when there isn't proper cleanup and can require manual steps which makes the process very slow.